### ISSUE TITLE
__new__ covariant type variable

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3839]
**Describe the bug**
```python
from typing import Generic, TypeVar

T = TypeVar("T", covariant=True)


class Box(Generic[T]):
    def __init__(self, value: T) -> None: # No error as expected.
        ...

class Box2(Generic[T]):
    def __new__(cls, value: T) -> "Box2[T]": ... # Error here of `Covariant type variable cannot be used in parameter type`
```

**Expected behavior**
I would expect new similar to init to have exception for covariant parameter rule. Mypy does not report an error here and I'm unsure how it can be unsafe for new, but safe for init.

**VS Code extension or command-line**
pyright cli 1.1.267

**Comment by erictraut** [comment]
I think it's reasonable to exempt `__new__` from this parameter variance check. As you point out, `__init__` is already exempted. 

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.268, which I just published. It will also be included in a future version of pylance.
